ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -419,115 +398,36 @@ public boolean hasSucceeded() {
-      return true;
-    }
-    // TODO: When helix external view doesn't include OFFLINE replicas, the totalOfflineReplicaCount is zero.
-    // FIX IT.
-    if (hasFailedOnCrossColoNotFound() && allReplicaCount - totalNotFoundCount >= replicaSuccessTarget
-        && totalOfflineReplicaCount > 0) {
-      logger.info(
-          "Terminating {} on {} due to disk down count and total Not_Found count from eligible replicas and some "
-              + "other replicas being unavailable. CrossColoEnabled: {}, DiskDownCount: {}, TotalNotFoundCount: {}, "
-              + "TotalReplicaCount: {}, replicaSuccessTarget: {}, OfflineReplicaCount: {}, allReplicaCount: {} {} "
-              + "replicasByState = {}",
-          routerOperation, partitionId, crossColoEnabled, diskDownCount, totalNotFoundCount, totalReplicaCount,
-          replicaSuccessTarget, totalOfflineReplicaCount, allReplicaCount, getBlobIdLog(), allDcReplicasByState);
-      routerMetrics.failedMaybeDueToTotalOfflineReplicasCount.inc();
+      routerMetrics.failedOnTotalNotFoundCount.inc();
       return true;
     }
+
     return false;
   }
 
   @Override
-  public boolean hasFailedOnNotFound() {
-    if (routerOperation == RouterOperation.PutOperation) {
-      return false;
-    }
-    if (hasFailedOnOriginatingDcNotFound()) {
-      logger.info(
-          "Terminating {} on {} due to Not_Found failure. Originating Not_Found count: {}, failure threshold: {},"
-              + "originatingDcOfflineReplicaCount: {}, originatingDcNameTotalReplicaCount: {},"
-              + "replicaSuccessTarget: {}, allReplicaCount: {} {}", routerOperation.name(), partitionId,
-          originatingDcNotFoundCount, originatingDcNotFoundFailureThreshold, originatingDcOfflineReplicaCount,
-          originatingDcTotalReplicaCount, replicaSuccessTarget, allReplicaCount, getBlobIdLog());
-      routerMetrics.failedOnOriginatingDcNotFoundCount.inc();
-      return true;
-    }
-    // To account for GET operation, the threshold should be  >= totalReplicaCount - (success target - 1)
-    // Right now, this only applies for replica only partitions and may not be completely accurate if there are
-    // failures responses other than not found.
-    if (hasFailedOnCrossColoNotFound()) {
-      logger.info(
-          "Terminating {} on {} due to disk down count and total Not_Found. CrossColoEnabled: {}, DiskDownCount: {},"
-              + "TotalNotFoundCount: {}, TotalReplicaCount: {}, replicaSuccessTarget: {}, OfflineReplicaCount: {},"
-              + "allReplicaCount: {} {} replicasByState = {}", routerOperation, partitionId, crossColoEnabled, diskDownCount,
-          totalNotFoundCount, totalReplicaCount, replicaSuccessTarget, totalOfflineReplicaCount, allReplicaCount,
-          getBlobIdLog(), allDcReplicasByState);
-      routerMetrics.failedOnTotalNotFoundCount.inc();
-      return true;
-    }
-    return false;
+  public boolean hasFailedOnUnavailability() {
```

Comment:
I think we need change this name "hasFailedOnUnavailability".
The operation is not necessary failed on unavailability. Just we have some replicas are un-available. 
Does "hasSomeUnavailability" or something like that less confusing?

Three originating replicas, replica 1 is leader replica and returns deleted, replica 2 is standby and returns deleted, replica 3 is bootstrap, when we enable check_all_originating_replica, this function will return true.
But we don't fail on unavailability. The error status should be deleted.

Also we may have an issue in the GetOperation.

Reply:
Sure. Changed the name to `hasSomeUnavailability` and updated to call `setChunkException` instead of `exception = e `for unavailability error.